perllib/FixMyStreet/App/Controller/Report.pm

```diff
@@ -660,6 +660,43 @@ sub map :Chained('id') :Args(0) {
     $c->res->body($image->{data});
 }
 
+sub confirmation : Path('confirmation') : Args(1) {
+    my ( $self, $c, $id ) = @_;
+
+    # First of all check that the report ID is valid
+    my $report = FixMyStreet::DB->resultset('Problem')->find( { id => $id } );
+    unless ( $report ) {
+        $c->detach( '/page_error_404_not_found', [] );
+    }
+
+    # Now verify the token we've been given is correct
+    my $token_param = $c->get_param('token') || "";
+    unless ( $token_param eq $report->confirmation_token ) {
+        $c->detach( '/page_error_404_not_found', [] );
+    }
+
+    # If the token is valid but expired then may as well be helpful and bounce
+    # the user to report page rather than 404.
+    # (NB the report may still be unconfirmed, but end result is the same - a 404)
+    my $cutoff = DateTime->now()->subtract( minutes => 1 );
+    my $timestamp = $report->confirmed || $report->created;
+    if ( $timestamp < $cutoff ) {
+        return $c->res->redirect($report->url);
```

Comment:
There's an issue (not caused by the fixup which is fine, I didn't notice the test failure before, but can see them at https://github.com/mysociety/fixmystreet/actions/workflows/coverage.yml?query=branch%3Aredirect-after-report-creation ) with it sometimes, but not always, redirecting to /report/ID rather than confirmation on a report that isn't viewable (eg. two-tier for Hart in report_new.t) and so giving a 404. My only thought is because coverage tests run more slowly than normal, the minute cut-off has been (sometimes, not always) reached and so the thing is trying to redirect but because these are cross-domain reports, that doesn't work. If that's the case, replacing this with the same as the confirm problem template of `c.cobrand.relative_url_for_report( report )` joined to `report.url` should work.

Reply:
Oops, nice catch!